src/display/api.js

```diff
@@ -398,9 +398,16 @@ function getDocument(src) {
       if (task.destroyed) {
         throw new Error("Loading aborted");
       }
+      if (worker.destroyed) {
+        throw new Error("Worker was destroyed");
+      }
 
-      const workerIdPromise = _fetchDocument(worker, fetchDocParams);
-      const networkStreamPromise = new Promise(function (resolve) {
+      const workerIdPromise = worker.messageHandler.sendWithPromise(
+        "GetDocRequest",
+        docParams,
+        data ? [data.buffer] : null
+      );
+      const networkStreamPromise = new Promise(resolve => {
```

Comment:
`resolve` inside this Promise constructor is always called synchronously -- is it actually needed to have it as a promise?

Reply:
As with many other things that were once necessary, other changes have removed the need for this to be async :-)